reconcile/openshift_base.py

```diff
@@ -294,6 +297,14 @@ def realize_data(dry_run, oc_map, ri,
         enable_deletion = False
 
     for cluster, namespace, resource_type, data in ri:
+        if ri.has_error_registered(cluster=cluster):
+            msg = (
+                "[{}] skipping realize_data for "
+                "cluster with errors"
+            ).format(cluster)
+            logging.error(msg)
+            continue
+
         # desired items
         for name, d_item in data['desired'].items():
             c_item = data['current'].get(name)
```

Comment:
there's a call for `ri.register_error()` at line 377 below. Should that one include the cluster now?

Reply:
Wondering if we should make the cluster mandatory for the `register_error()`. We could call `has_error_registered()` without the cluster info when needed. Let's see how it goes and use a different PR if we see fit.

Reply:
i thought about it as well and decided to go with minimal required changes.
`ri.register_error` with a provided cluster is needed for this stage (`realize_data`) and is not (currently) needed for anything after this phase. i plan to add `cluster` to other error registrations in a future PR.